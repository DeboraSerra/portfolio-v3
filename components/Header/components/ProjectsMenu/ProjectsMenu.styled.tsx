import styled from "styled-components";

export const Main = styled.div`
  position: absolute;
  z-index: 100;
  top: 40px;
  display: flex;
  flex-direction: column;
  background-color: ${(props) => props.theme.bg.secondary};
  padding: 12px;
  box-shadow: 0 12px 24px ${(props) => props.theme.bg.input};

  color: ${(props) => props.theme.text.secondary};

  & a {
    margin-bottom: 8px;

    &:last-of-type {
      margin-bottom: 0;
    } // &:last-of-type
    &:hover {
      color: ${(props) => props.theme.text.tertiary};
      text-decoration: underline;
    } // &:hover
  } // & a

  @media only screen and (max-width: 576px) {
    top: 28px;
  } // @media only screen and (max-width : 576px)
`;
