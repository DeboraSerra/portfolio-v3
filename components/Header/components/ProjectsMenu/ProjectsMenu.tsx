import { NextPage } from "next";
import Link from "next/link";

import * as S from "./ProjectsMenu.styled";

interface Props {}

const ProjectsMenu: NextPage<Props> = () => {
  return (
    <S.Main>
      <Link href="/fundaments">Fundaments</Link>
      <Link href="/front-end">Front end</Link>
      <Link href="/back-end">Back end</Link>
      <Link href="/computer-science">Computer science</Link>
      <Link href="/personal">Personal</Link>
    </S.Main>
  );
};

export default ProjectsMenu;
