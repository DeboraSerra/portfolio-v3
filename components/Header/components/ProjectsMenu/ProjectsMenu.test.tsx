import { render } from "@testing-library/react";
import TProvider from "../../../helpers/ThemeProvider";
import ProjectsMenu from "./ProjectsMenu";

describe("Test the ProjectsMenu component", () => {
  it("Should match the snapshot", () => {
    const { asFragment } = render(
      <TProvider>
        <ProjectsMenu />
      </TProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
