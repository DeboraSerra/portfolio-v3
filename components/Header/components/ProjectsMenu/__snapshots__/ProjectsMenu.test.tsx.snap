// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the ProjectsMenu component Should match the snapshot 1`] = `
<DocumentFragment>
  .c0 {
  position: absolute;
  top: 64px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #f1b2b2;
  padding: 12px;
  box-shadow: 0 12px 24px #45496a6f;
  color: #2f3037;
}

.c0 a {
  margin-bottom: 8px;
}

.c0 a:last-of-type {
  margin-bottom: 0;
}

.c0 a:hover {
  color: #525560;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

@media only screen and (max-width:576px) {
  .c0 {
    top: 40px;
  }
}

<div
    class="c0"
  >
    <a
      href="/fundaments"
    >
      Fundaments
    </a>
    <a
      href="/front-end"
    >
      Front end
    </a>
    <a
      href="/back-end"
    >
      Back end
    </a>
    <a
      href="/computer-science"
    >
      Computer science
    </a>
    <a
      href="/personal"
    >
      Personal
    </a>
  </div>
</DocumentFragment>
`;
